zebrad/src/components/mempool/storage.rs

```diff
@@ -138,6 +163,11 @@ impl Storage {
             .collect()
     }
 
+    /// Returns the set of [`Transaction`]s in the mempool.
+    pub fn transactions_all(&self) -> Vec<UnminedTx> {
```

Comment:
This requires copying the entire storage, and it seems it can be avoided without any downsides:

```suggestion
    pub fn transactions_all(&self) -> impl Iterator<Item = &UnminedTx>
```

I realized it was already like this before this PR, but it seems a good opportunity to optimize it, if you agree.

Reply:
Thanks for catching this!

Reply:
I might apply this optimisation in PR #2833, to avoid merge conflicts.